src/pkgcheck/checks/metadata.py

```diff
@@ -1627,3 +1627,67 @@ def feed(self, pkg):
         for unpackers, filenames in missing_unpackers.items():
             yield MissingUnpackerDep(
                 str(pkg.eapi), sorted(filenames), sorted(unpackers), pkg=pkg)
+
+
+class VirtualWithSingleProvider(results.PackageResult, results.Warning):
```

Comment:
_Maybe_ we should suppress this if the package is in `package.deprecated`? 

Not sure if it's worth the effort.

Reply:
done